src/core/hooks/use-preset.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect } from 'react'
+
+import dirInfoStore from '@store/dirinfo'
+import { applyPreset } from '@pages/filter/refiner/components/solution-control-refiner/solution-control-refiner.utils'
```

Comment:
i think, we should do it in store level, preset store should observe urls and apply preset from it. I working on this base functionality now. Maybe we can wait a bit with this change